web3wallet (#535)
WalletConnect/WalletConnectKotlinV2: web3/wallet/ReadMe.md

# **WalletConnect Sign - Kotlin**

Kotlin implementation of web3wallet for Android applications.

![Maven Central](https://img.shields.io/maven-central/v/com.walletconnect/web3wallet)

## Requirements

* Android min SDK 23
* Java 11

## Documentation and usage

* [web3wallet installation](TBD)
* [web3wallet usage](TBD)

&nbsp;

## Installation

root/build.gradle.kts:

```gradle
allprojects {
 repositories {
    mavenCentral()
    maven { url "https://jitpack.io" }
 }
}
```

app/build.gradle.kts

```gradle
implementation("com.walletconnect:web3wallet:release_version")
```

&nbsp;

## Migration guide from SignClient and AuthClient to Web3Wallet

> Web3Wallet SDK introduces a new interface for all the wallet users that wraps the Sign and Auth clients internally. Only one dependence is needed to

Reviewer: ```suggestion
Web3Wallet SDK introduces a new interface for all the wallet users that wraps the Sign and Auth clients internally. Only one dependency is needed to
```

enable new use cases for wallets. Down below you can find a migration guide for Sing and Auth clients.

## Migration from SignClient

### Initialization

```kotlin
//CoreClient
CoreClient.initialize(relayServerUrl, connectionType, application, metaData) { error -> }

//SignClient
val initParams = Sign.Params.Init(core = CoreClient)
SignClient.initialize(initParams) { error -> }

//Web3Wallet
val initParams = Wallet.Params.Init(core = CoreClient)
Web3Wallet.initialize(initParams) { error -> }
```

### Pair with a Dapp

```kotlin
//CoreClient
val pairingParams = Core.Params.Pair(pairingUri)
CoreClient.Pairing.pair(pairingParams) { error -> }

//Web3Wallet
val pairingParams = Wallet.Params.Pair(pairingUri)
Web3Wallet.pair(pairingParams) { error -> }
```

### Approve a session

```kotlin
//SignClient
val approveParams: Sign.Params.Approve = Sign.Params.Approve(proposerPublicKey, namespaces)
SignClient.approveSession(approveParams) { error -> }

//Web3Wallet
val approveParams = Wallet.Params.SessionApprove(proposerPublicKey namespaces)
Web3Wallet.approveSession(approveProposal) { error -> }
```

### Reject a session

```kotlin
//SignClient

> val rejectParams = Sign.Params.Reject = Reject(proposerPublicKey, rejectionReason, rejectionCode)

Reviewer: ```suggestion
val rejectParams: Sign.Params.Reject = Sign.Params.Reject(proposerPublicKey, rejectionReason, rejectionCode)
```

# **WalletConnect Web3Wallet - Kotlin**

Kotlin implementation of Web3Wallet for Android applications.

![Maven Central](https://img.shields.io/maven-central/v/com.walletconnect/web3wallet)

## Requirements

* Android min SDK 23
* Java 11

## Documentation and usage

* [Web3Wallet installation](TBD)
* [Web3Wallet usage](TBD)

## Installation

root/build.gradle.kts:

```gradle
allprojects {
 repositories {
    mavenCentral()
    maven { url "https://jitpack.io" }
 }
}
```

app/build.gradle.kts

```gradle
implementation("com.walletconnect:web3wallet:release_version")

> ```

Reviewer 2: Should we also show an example using the BOM?

Author: I can add it, to keep it consistent will add it to all ReadMes

&nbsp;

## Migration guide from SignClient and AuthClient to Web3Wallet

> Web3Wallet SDK introduces a new interface for all the wallet users that wraps the Sign and Auth clients internally. Only one dependency is needed to

Reviewer 2: ```suggestion
Web3Wallet SDK introduces a new interface for all wallets that wraps the Sign and Auth clients internally. Only one dependency is needed to
```

//CoreClient
CoreClient.initialize(relayServerUrl, connectionType, application, metaData) { error -> }

//AuthClient
val initParams = Auth.Params.Init(CoreClient)
AuthClient.initialize(init = Auth.Params.Init(core = CoreClient)) { error -> }

//Web3Wallet
val initParams = Wallet.Params.Init(core = CoreClient)
Web3Wallet.initialize(initParams) { error -> }
```

### Authentication

```kotlin
//AuthClient
val signature = CacaoSigner.sign(message, privateKey, SignatureType.EIP191)
AuthClient.respond(Auth.Params.Respond.Result(requestId, signature, issuer)) { error -> }

//Web3Wallet
val signature = CacaoSigner.sign(message, privateKey, SignatureType.EIP191)
Web3Wallet.respond(Wallet.Params.AuthRequestResponse(requestId, signature, issuer)) { error -> }
```

### Message formatting

```kotlin
//AuthClient
val payloadParams: Auth.Params.PayloadParams = //PayloadParams received in the onAuthRequest callback
val issuer = //MUST be the same as send with the respond methods and follows: https://github.com/w3c-ccg/did-pkh/blob/main/did-pkh-method-draft.md
val formatMessage = Auth.Params.FormatMessage(payloadParamspayloadParams, issuer)
AuthClient.formatMessage(formatMessage)

//Web3Wallet

> val payloadParams: Wallwt.Model.PayloadParams  = //PayloadParams received in the onAuthRequest callback

Reviewer: ```suggestion
val payloadParams: Wallet.Model.PayloadParams  = //PayloadParams received in the onAuthRequest callback
```
